webui/react/src/hooks/useAuthCheck.ts

```diff
@@ -17,7 +18,21 @@ const useAuthCheck = (): (() => void) => {
 
   useEffect(() => {
     const checkAuth = async (signal: AbortSignal): Promise<void> => {
-      const authToken = getCookie(AUTH_COOKIE_KEY) || globalStorage.authToken;
+      /*
+       * Check for an auth token in the cookie from SSO and
+       * update the storage token and the api to use the cookie token.
+       */
+      const cookieToken = getCookie(AUTH_COOKIE_KEY);
+      if (cookieToken) {
+        globalStorage.authToken = cookieToken;
+        updateDetApi({ apiKey: 'Bearer ' + cookieToken });
+      }
+
+      /*
+       * If a cookie token is not found, use the storage token if applicable.
+       * Proceed to verify user only if there is an auth token.
+       */
+      const authToken = cookieToken || globalStorage.authToken;
```

Comment:
nit: we probably don't need `cookieToken` here. if it was available it would have overridden `global.authToken`. 

I'm thinking whether this could interfere with multi-cluster setup

question: would this work? for line 25
`      if (cookieToken && !globalStorage.authToken) {`